<launch>
    <arg name="robot" default="fsquash" />
    <arg name="output" default="screen" />
  
    <!-- Squash -->
    <arg if="$(eval arg('robot') == 'squash')" name="pack" default="p5" />
    <arg if="$(eval arg('robot') == 'squash')" name="c" default="squash" />
    <arg if="$(eval arg('robot') == 'squash')" name="robot_name" default="r6" />
    <arg if="$(eval arg('robot') == 'squash')" name="self_strike" default="squash" />
  
    <!-- Squash -->
    <arg if="$(eval arg('robot') == 'fsquash')" name="platform" default="squash" />
    <arg if="$(eval arg('robot') == 'fsquash')" name="robot_name" default="r6" />
    <arg if="$(eval arg('robot') == 'fsquash')" name="self_strike" default="squash" />
    <arg if="$(eval arg('robot') == 'fsquash')" name="pack" default="p9" />
  
    <arg name="ROSCONSOLE_FORMAT" default="${severity} ${time} ${node}: ${message}" />
    <env name="ROSCONSOLE_FORMAT" value="$(arg ROSCONSOLE_FORMAT)" />
  
    <group ns="$(arg robot_name)">
      <remap from="/tf" to="tf" />
      <remap from="/tf_static" to="tf_static" />
  
        <!-- ONLINE Treaining node -->
        <node pkg="ros_scnn_tm" type="scnn_online_trainer.py" name="scnn_online_trainer" clear_params="true" output="screen">
          <!-- Default learning params, non-ros -->
          <param name="learning_param_file" value="$(find nve_startup)/config/squash/squash_online_learning_params.yaml" />
          
          <!-- ROS specific params -->
          <param name="main_rate" value="1.0" />
          <param name= "min_pose_dist" value="1.0" />
          <param name="~model_out_dir" value="/data/debug/test_online_trainer/incemental_online_models/" />
          

          
          <!-- Remaps for topics and intemediate out directoriesf -->
          <remap from="~new_training_data" to="/$(arg robot_name)/data_fuser_node/new_training_data" />

        </node>
  
    </group>
  </launch>
  